mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_api.py

```diff
@@ -0,0 +1,138 @@
+import requests
+from requests import Response
+from enum import Enum
+
+
+class TripAdvisorAPICall(Enum):
+    SEARCH_LOCATION = 1
+    LOCATION_DETAILS = 2
+    PHOTOS = 3
+
+
+class TripAdvisorAPI:
+    """A class for checking the connection to the TripAdvisor Content API and making requests.
+
+    Attributes:
+        api_key (str): The unique API key to access Tripadvisor content.
+    """
+
+    def __init__(self, api_key):
+        self.api_key = api_key
+
+    def connectTripAdvisor(self):
+        """
+        Check the connection with TripAdvisor
+        """
+        url = "https://api.content.tripadvisor.com/api/v1/location/search?language=en&key={api_key}&searchQuery={searchQuery}".format(
+            api_key=self.api_key, searchQuery="London"
```

Comment:
Does this limit the search to London?

Reply:
Should have explained more about my custom wrapper, this part of the code just basically check whether there is a healthy connection with TripAdvisor platform (StatusResponse).

It's for _check_connection()_ function in APIHandler class.

Reply:
Hmm.. But, you are using an object of this class in other parts of your code as well, right? Specifically, when making requests? For example, in `call_tripadvisor_searchlocation_api()`? Wouldn't that limit the search only to the London area?

Reply:
It doesn't. It is a bit complicated, and I should have documented that part better. I used that part of the code only to check the connection with TripAdvisor. 

If there are 4xx errors (client used incorrect API key) or 5xx errors (server problem), it notifies the users in MindsDB.

In _"call_tripadvisor_searchLocation_api()"_, I used only functions **"make_request"**, **"process_query"** and **"location_details"** in custom wrapper (_tripadvisor_api.py_) for handling user requests.

Reply:
Ah, right. Understood. Shall we then change the name of the function to make it a little clearer? Maybe `checkTripAdvisorConnection()` or anything else that is a little more descriptive of its purpose?